matsim/src/main/java/org/matsim/core/mobsim/qsim/jdeqsimengine/JDEQSimEngine.java

```diff
@@ -31,12 +32,21 @@
     private int numberOfAgents = 0;
     private AgentCounter agentCounter;
 
+    /**
+     * this must be initialized before starting the simulation! mapping:
+     * key=linkId used to find a road corresponding to a link
+     */
+    protected final HashMap<Id<Link>, Road> allRoads;
```

Comment:
Another reason to not do this in another place - now there's more than a single place where this `allRoads` is being managed..

Reply:
These are different allRoads (JDEQSimEngine and JDEQSimulation). But this shows that we had 2 places where the single instance of allRoads was inited. Probably this was intentionally. Though I didn't find usage of that JDEQSimEngine.